javascript/src/parser.ts

```diff
@@ -58,25 +65,22 @@ export class Parser {
    */
   parseBoundaries(sentence: string): number[] {
     const result = [];
-    const baseScore =
-      -0.5 *
-      [...this.model.values()]
-        .map(group => [...group.values()])
-        .flat()
-        .reduce((prev, curr) => prev + curr, 0);
 
     for (let i = 1; i < sentence.length; i++) {
-      let score = baseScore;
+      let score = this.baseScore;
       /* eslint-disable */
       score += this.model.get('UW1')?.get(sentence.substring(i - 3, i - 2)) || 0;
       score += this.model.get('UW2')?.get(sentence.substring(i - 2, i - 1)) || 0;
       score += this.model.get('UW3')?.get(sentence.substring(i - 1, i)) || 0;
+      if (score > 0) { result.push(i); continue; }
```

Comment:
Items cannot be bailed out earlier because the score may get negative in the later summations. The same applies to L79 and 83.

Reply:
Oh; I see. That's why tests are always important as well.

I can add a commit to add a comment that emphasizes/mentions that.

Thanks Shuhei.